### ISSUE TITLE
Visual indicator that a file filter is in effect

### ISSUE THREAD

**Issue by vidartf** [jupyterlab/jupyterlab#10911]
### Problem

Currently, users might have a file filter active in the file browser, and not realize it (particularly since it is retained even after browsing into a different folder, which is probably unexpected). They will then be annoyed when they cannot find their files, and instead only see empty folders (with a few files if they happen to match the filter).

### Proposed Solution

A number of proposals might be considered independent of each other:
1. Add some clear visual queue that a filter is currently in effect. Either by a border, overlay, or icon (doing some research on what other applications use would be good).
2. Clear the file filter when you browse to a different folder. Very rarely does the filter you applied in one folder translate well to another, and very rarely is that what you would want.
3. Adding a clear button to the input ("X" button) to quickly clear the filter. Most filters are temporary in nature, e.g. used to help find a file in a folder with a lot of entries. Once the file has been found, in most cases the filter is no longer necessary.

Note that proposals 1 and 3 might be combined, similarly to what vscode does:

With a filter of "rc" (on both folders and files in vscode's case), a overlay is shown, and matches are highlighted:
![image](https://user-images.githubusercontent.com/510760/130767832-660cfb95-9a69-4c42-a95d-c21d04aee047.png)

On mouse hover on overlay, a clear button is shown (in addition to other filter options):
![image](https://user-images.githubusercontent.com/510760/130767855-c74fa5aa-21af-438d-a19f-7a2d112290a1.png)

**Comment by krassowski** [comment]
Also, we already have a quick find option (just start typing in the listing), and I believe that the filter field is redundant with respect to that. I think that the quick find on listing is the thing that should receive an indicator (to show what prefix is being used for quick find). I will follow up after hours.

**Comment by vidartf** [comment]
@krassowski Yes, I do not see that the quick-find functionality was mentioned in the [original issue](https://github.com/jupyterlab/jupyterlab/issues/8157) or [PR](https://github.com/jupyterlab/jupyterlab/pull/8615).

**Comment by treyhunner** [comment]
I teach many students who use JupyterLab as their Python environment. These are first-time Python and Jupyter users. The biggest issue we tend to run into is an accidental filter on the file browser. This typically happens when students type a search in to find the folder they're looking for, double click the folder name, and then see that the folder appears to be unexpectedly empty. The folder isn't *actually* empty, but the filter does not clear when navigating into a subdirectory in the file browser so what the (now very narrow) filter makes the folder appear empty.

Solution 2 proposed in the original issue would likely solve this confusion entirely:

> Clear the file filter when you browse to a different folder. Very rarely does the filter you applied in one folder translate well to another, and very rarely is that what you would want.

It looks like solution 3 has been implemented already. I hope that solution will eventually be implemented as well.
